src/app/pages/client-reviews/client-reviews.component.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { HttpClient, HttpClientModule } from '@angular/common/http';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { httpClientSpy, toastServiceSpy, reviewsSpy, resetSpies } from 'src/app/helpers/spies';
+import { of, throwError } from 'rxjs';
+import { ToastrService } from 'ngx-toastr';
+import { RouterTestingModule } from '@angular/router/testing';
+import { ClientReviewsComponent } from './client-reviews.component';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { NgxSpinnerModule } from 'ngx-spinner';
+
+describe('ClientReviewsComponent', () => {
+  let component: ClientReviewsComponent;
+  let fixture: ComponentFixture<ClientReviewsComponent>;
+  let debugElement: DebugElement;
+  beforeAll(() => resetSpies([reviewsSpy]));
+  afterEach(() => resetSpies([reviewsSpy]));
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ClientReviewsComponent],
+      imports: [HttpClientModule, NgxSpinnerModule,
+RouterTestingModule.withRoutes([{ path: '**', component: ClientReviewsComponent }, ]),
+      ],
+      providers: [
+        {
+          provide: ClientReviewsService,
+          useValue: reviewsSpy
+        },
+        {
+          provide: HttpClient,
+          useValue: httpClientSpy
+        },
+
+        {
+          provide: ToastrService,
+          useValue: toastServiceSpy
+        },
+      ]
+    })
+      .compileComponents();
+  }));
+
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ClientReviewsComponent);
+    component = fixture.componentInstance;
+    reviewsSpy.getReviews.and.returnValue(of(mockReviewsResponse));
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+  it('should go to next page on button click', () => {
+    debugElement = fixture.debugElement.query(By.css('.next'));
+    debugElement.triggerEventHandler('click', null);
+    expect(reviewsSpy.getReviews).toHaveBeenCalled();
+  });
+
+  it('should go to previous page on button click', () => {
+    debugElement = fixture.debugElement;
+    debugElement.query(By.css('.prev')).triggerEventHandler('click', null);
+    expect(reviewsSpy.getReviews).toHaveBeenCalled();
+  });
+
+  it('should set the components next property when reviews exist', () => {
+    const url = 'http://127.0.0.1:8000/api/v1/auth/1/reviews';
+    const response = {
+      count: 0,
+      next: 'http://127.0.0.1:8000/api/v1/auth/1/reviews/?limit=10&offset=10',
+      previous: 'http://127.0.0.1:8000/api/v1/auth/1/reviews/?limit=10&offset=10',
+      results: []
+    };
+
+
+    reviewsSpy.getReviews.and.returnValue(of(response));
+    component.fetchReviews(url);
+    expect(component.next).toEqual(response.next);
+    expect(component.prev).toEqual(response.previous);
+  });
+  it('should throw a toast error', () => {
+    const url = 'http://127.0.0.1:8000/api/v1/auth/1/reviews';
```

Comment:
You are declaring this url twice, I think it would be better if you declare it once, and access it from wherever you need it

Reply:
Done